src/logger.cr

```diff
@@ -7,32 +7,29 @@ class Logger
   alias FilterType = Filter | Severity | (Entry -> Bool)
   alias EmitterType = Emitter | (Entry -> Nil)
 
-  module Base
-    abstract def component : String
-    abstract def filter : FilterType?
-    abstract def emitters : Array(EmitterType)
-
-    def log(entry : Entry)
-      case ff = filter
-      when Filter, Proc(Entry, Bool)
-        return unless ff.call(entry)
-      when Severity
-        return unless entry.severity >= ff
-      end
-      emitters.each &.call(entry)
+  property component : String
+  property filter : FilterType?
+  property emitters : Array(EmitterType)
```

Comment:
Why did the ivar accessors became properties? I don't think they need to be mutable. `getter` should be enough.

Reply:
I made `filter` a property since `Logger::ROOT.filter = ...` needs to be possible, and then brought the rest along for consistency. But I agree immutability would be better in other cases. I think that's an argument in favor of making `Logger.get` return a class that is thinner than `Logger`.

`component` can be made immutable, in any case. `emitters` doesn't really matter since the array will always be mutable.

Reply:
Ah yes, I see the last commit. Yeah, when we're not using thin wrappers, `filter` should probably be mutable. But `component` and `emitters` do not.